WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -32,18 +32,30 @@ public LabelSelectionViewModel(Money targetAmount)
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] AutoSelectPockets()
+	public Pocket[] AutoSelectPockets(SmartLabel recipient)
 	{
 		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 		var unknownPockets = NonPrivatePockets.Except(knownPockets).ToArray();
 		var privateAndUnknownPockets = _allPockets.Except(knownPockets).ToArray();
 		var privateAndKnownPockets = _allPockets.Except(unknownPockets).ToArray();
+		var knownByRecipientPockets = knownPockets.Where(x => x.Labels.Any(label => recipient.Labels.Contains(label))).ToArray();
+		var onlyKnownByRecipientPockets = knownByRecipientPockets.Where(x => x.Labels.All(label => recipient.Labels.Contains(label))).ToArray();
```

Comment:
Nit: var name should be `onlyKnownByRecipientPocket` since there is only one pocket.

Reply:
Yes, it could only contain one pocket, but that variable is an array, so that is why I went with the plural form.
I can make it to be only a `Pocket` object, but the method returns an array so it should be put back into an array when returning it. I found the current solution simple.

Reply:
I get it, no need I was just talking about the var name. Feel free to ignore.